### ISSUE TITLE
stack overflow when printing ErrorNoCluster

### ISSUE THREAD

**Issue by nbari** [grafana/tanka#238]
Hi, I am trying to install `loki` following this guide: https://github.com/grafana/loki/blob/master/docs/installation/tanka.md

But when trying to deploy by running: 

     tk apply environments/loki 

This is the output:

    runtime: goroutine stack exceeds 1000000000-byte limit
    runtime: sp=0xc0290c0350 stack=[0xc0290c0000, 0xc0490c0000]
    fatal error: stack overflow

I am using:

    $ tk --version
    tk version v0.8.0-9-g9fa4f27
   
     $ jb --version
    dev

my `environments/loki/main.jsonnet|` file looks like this:

```
local gateway = import 'loki/gateway.libsonnet';
local loki = import 'loki/loki.libsonnet';
local promtail = import 'promtail/promtail.libsonnet';

loki + promtail + gateway {
    _config+:: {
        namespace: 'loki',
        htpasswd_contents: 'loki:$apr1$iVatTW2t$FArAQxAxdjabx0faF8KVY0',

        // GCS variables remove if not using gcs
        storage_backend: 'gcs',
        gcs_bucket_name: 'test-loki',

        promtail_config+: {
            clients: [{
                scheme:: 'http',
                hostname:: 'gateway.%(namespace)s.svc' % $._config,
                username:: 'loki',
                password:: 'password',
                container_root_path:: '/var/lib/docker',
            }],
        },

        replication_factor: 3,
        consul_replicas: 1,
    },
}
```

The Kubernetes cluster is running on GPC 

Full is the full output:

```
runtime stack:
runtime.throw(0x1342476, 0xe)
        /usr/local/Cellar/go/1.14/libexec/src/runtime/panic.go:1112 +0x72
runtime.newstack()
        /usr/local/Cellar/go/1.14/libexec/src/runtime/stack.go:1034 +0x6ce
runtime.morestack()
        /usr/local/Cellar/go/1.14/libexec/src/runtime/asm_amd64.s:449 +0x8f

goroutine 1 [running]:
runtime.heapBitsSetType(0xc0077b51e0, 0xd0, 0xc8, 0x1328fc0)
        /usr/local/Cellar/go/1.14/libexec/src/runtime/mbitmap.go:947 +0xa4d fp=0xc029700360 sp=0xc029700358 pc=0x10159fd
runtime.mallocgc(0xd0, 0x1328fc0, 0x1, 0x0)
        /usr/local/Cellar/go/1.14/libexec/src/runtime/malloc.go:1066 +0x53e fp=0xc029700400 sp=0xc029700360 pc=0x100bede
runtime.newobject(0x1328fc0, 0xc0297004d8)
        /usr/local/Cellar/go/1.14/libexec/src/runtime/malloc.go:1165 +0x38 fp=0xc029700430 sp=0xc029700400 pc=0x100c528
fmt.glob..func1(0x15d8940, 0x0)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:132 +0x2d fp=0xc029700450 sp=0xc029700430 pc=0x10e120d
sync.(*Pool).Get(0x15d8940, 0xc00004d000, 0x181a108)
        /usr/local/Cellar/go/1.14/libexec/src/sync/pool.go:148 +0xa6 fp=0xc029700498 sp=0xc029700450 pc=0x106bf36
fmt.newPrinter(0x0)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:137 +0x31 fp=0xc0297004c0 sp=0xc029700498 pc=0x10d2fd1
fmt.Sprintf(0x135e346, 0x53, 0xc029700550, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:218 +0x26 fp=0xc029700518 sp=0xc0297004c0 pc=0x10d3596
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b22a0, 0x12e0da0, 0xc0077b22a0)
        <autogenerated>:1 +0xa4 fp=0xc029700570 sp=0xc029700518 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b5110, 0x73, 0xc029700801)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc0297007e0 sp=0xc029700570 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b5110, 0x12e0da0, 0xc0077b22a0, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029700878 sp=0xc0297007e0 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b5110, 0x135e346, 0x53, 0xc0297009f0, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029700960 sp=0xc029700878 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc0297009f0, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc0297009b8 sp=0xc029700960 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b2290, 0x12e0da0, 0xc0077b2290)
        <autogenerated>:1 +0xa4 fp=0xc029700a10 sp=0xc0297009b8 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b5040, 0x73, 0xc029700d01)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029700c80 sp=0xc029700a10 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b5040, 0x12e0da0, 0xc0077b2290, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029700d18 sp=0xc029700c80 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b5040, 0x135e346, 0x53, 0xc029700e90, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029700e00 sp=0xc029700d18 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029700e90, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc029700e58 sp=0xc029700e00 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b2280, 0x12e0da0, 0xc0077b2280)
        <autogenerated>:1 +0xa4 fp=0xc029700eb0 sp=0xc029700e58 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b4f70, 0x73, 0xc029701101)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029701120 sp=0xc029700eb0 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b4f70, 0x12e0da0, 0xc0077b2280, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc0297011b8 sp=0xc029701120 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b4f70, 0x135e346, 0x53, 0xc029701330, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc0297012a0 sp=0xc0297011b8 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029701330, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc0297012f8 sp=0xc0297012a0 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b2270, 0x12e0da0, 0xc0077b2270)
        <autogenerated>:1 +0xa4 fp=0xc029701350 sp=0xc0297012f8 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b4ea0, 0x73, 0xc029701601)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc0297015c0 sp=0xc029701350 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b4ea0, 0x12e0da0, 0xc0077b2270, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029701658 sp=0xc0297015c0 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b4ea0, 0x135e346, 0x53, 0xc0297017d0, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029701740 sp=0xc029701658 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc0297017d0, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc029701798 sp=0xc029701740 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b2260, 0x12e0da0, 0xc0077b2260)
        <autogenerated>:1 +0xa4 fp=0xc0297017f0 sp=0xc029701798 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b4dd0, 0x73, 0xc029701b01)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029701a60 sp=0xc0297017f0 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b4dd0, 0x12e0da0, 0xc0077b2260, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029701af8 sp=0xc029701a60 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b4dd0, 0x135e346, 0x53, 0xc029701c70, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029701be0 sp=0xc029701af8 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029701c70, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc029701c38 sp=0xc029701be0 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b2250, 0x12e0da0, 0xc0077b2250)
        <autogenerated>:1 +0xa4 fp=0xc029701c90 sp=0xc029701c38 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b4d00, 0x73, 0xc029701f01)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029701f00 sp=0xc029701c90 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b4d00, 0x12e0da0, 0xc0077b2250, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029701f98 sp=0xc029701f00 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b4d00, 0x135e346, 0x53, 0xc029702110, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029702080 sp=0xc029701f98 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029702110, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc0297020d8 sp=0xc029702080 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b2240, 0x12e0da0, 0xc0077b2240)
        <autogenerated>:1 +0xa4 fp=0xc029702130 sp=0xc0297020d8 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b4c30, 0x73, 0xc029702401)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc0297023a0 sp=0xc029702130 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b4c30, 0x12e0da0, 0xc0077b2240, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029702438 sp=0xc0297023a0 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b4c30, 0x135e346, 0x53, 0xc0297025b0, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029702520 sp=0xc029702438 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc0297025b0, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc029702578 sp=0xc029702520 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b2230, 0x12e0da0, 0xc0077b2230)
        <autogenerated>:1 +0xa4 fp=0xc0297025d0 sp=0xc029702578 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b4b60, 0x73, 0xc029702801)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029702840 sp=0xc0297025d0 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b4b60, 0x12e0da0, 0xc0077b2230, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc0297028d8 sp=0xc029702840 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b4b60, 0x135e346, 0x53, 0xc029702a50, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc0297029c0 sp=0xc0297028d8 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029702a50, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc029702a18 sp=0xc0297029c0 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b2220, 0x12e0da0, 0xc0077b2220)
        <autogenerated>:1 +0xa4 fp=0xc029702a70 sp=0xc029702a18 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b4a90, 0x73, 0xc029702d01)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029702ce0 sp=0xc029702a70 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b4a90, 0x12e0da0, 0xc0077b2220, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029702d78 sp=0xc029702ce0 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b4a90, 0x135e346, 0x53, 0xc029702ef0, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029702e60 sp=0xc029702d78 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029702ef0, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc029702eb8 sp=0xc029702e60 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b2210, 0x12e0da0, 0xc0077b2210)
        <autogenerated>:1 +0xa4 fp=0xc029702f10 sp=0xc029702eb8 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b49c0, 0x73, 0xc029703201)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029703180 sp=0xc029702f10 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b49c0, 0x12e0da0, 0xc0077b2210, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029703218 sp=0xc029703180 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b49c0, 0x135e346, 0x53, 0xc029703390, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029703300 sp=0xc029703218 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029703390, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc029703358 sp=0xc029703300 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b2200, 0x12e0da0, 0xc0077b2200)
        <autogenerated>:1 +0xa4 fp=0xc0297033b0 sp=0xc029703358 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b48f0, 0x73, 0xc029703601)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029703620 sp=0xc0297033b0 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b48f0, 0x12e0da0, 0xc0077b2200, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc0297036b8 sp=0xc029703620 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b48f0, 0x135e346, 0x53, 0xc029703830, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc0297037a0 sp=0xc0297036b8 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029703830, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc0297037f8 sp=0xc0297037a0 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b21f0, 0x12e0da0, 0xc0077b21f0)
        <autogenerated>:1 +0xa4 fp=0xc029703850 sp=0xc0297037f8 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b4820, 0x73, 0xc029703b01)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029703ac0 sp=0xc029703850 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b4820, 0x12e0da0, 0xc0077b21f0, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029703b58 sp=0xc029703ac0 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b4820, 0x135e346, 0x53, 0xc029703cd0, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029703c40 sp=0xc029703b58 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029703cd0, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc029703c98 sp=0xc029703c40 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b21e0, 0x12e0da0, 0xc0077b21e0)
        <autogenerated>:1 +0xa4 fp=0xc029703cf0 sp=0xc029703c98 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b4750, 0x73, 0xc029704001)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029703f60 sp=0xc029703cf0 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b4750, 0x12e0da0, 0xc0077b21e0, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029703ff8 sp=0xc029703f60 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b4750, 0x135e346, 0x53, 0xc029704170, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc0297040e0 sp=0xc029703ff8 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029704170, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc029704138 sp=0xc0297040e0 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b21d0, 0x12e0da0, 0xc0077b21d0)
        <autogenerated>:1 +0xa4 fp=0xc029704190 sp=0xc029704138 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b4680, 0x73, 0xc029704401)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029704400 sp=0xc029704190 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b4680, 0x12e0da0, 0xc0077b21d0, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029704498 sp=0xc029704400 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b4680, 0x135e346, 0x53, 0xc029704610, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029704580 sp=0xc029704498 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029704610, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc0297045d8 sp=0xc029704580 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b21c0, 0x12e0da0, 0xc0077b21c0)
        <autogenerated>:1 +0xa4 fp=0xc029704630 sp=0xc0297045d8 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b45b0, 0x73, 0xc029704901)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc0297048a0 sp=0xc029704630 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b45b0, 0x12e0da0, 0xc0077b21c0, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029704938 sp=0xc0297048a0 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b45b0, 0x135e346, 0x53, 0xc029704ab0, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029704a20 sp=0xc029704938 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029704ab0, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc029704a78 sp=0xc029704a20 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b21b0, 0x12e0da0, 0xc0077b21b0)
        <autogenerated>:1 +0xa4 fp=0xc029704ad0 sp=0xc029704a78 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b44e0, 0x73, 0xc029704d01)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029704d40 sp=0xc029704ad0 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b44e0, 0x12e0da0, 0xc0077b21b0, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029704dd8 sp=0xc029704d40 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b44e0, 0x135e346, 0x53, 0xc029704f50, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029704ec0 sp=0xc029704dd8 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029704f50, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc029704f18 sp=0xc029704ec0 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b21a0, 0x12e0da0, 0xc0077b21a0)
        <autogenerated>:1 +0xa4 fp=0xc029704f70 sp=0xc029704f18 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b4410, 0x73, 0xc029705201)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc0297051e0 sp=0xc029704f70 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b4410, 0x12e0da0, 0xc0077b21a0, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029705278 sp=0xc0297051e0 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b4410, 0x135e346, 0x53, 0xc0297053f0, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029705360 sp=0xc029705278 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc0297053f0, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc0297053b8 sp=0xc029705360 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b2190, 0x12e0da0, 0xc0077b2190)
        <autogenerated>:1 +0xa4 fp=0xc029705410 sp=0xc0297053b8 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b4340, 0x73, 0xc029705701)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029705680 sp=0xc029705410 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b4340, 0x12e0da0, 0xc0077b2190, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029705718 sp=0xc029705680 pc=0x10d6d74
fmt.(*pp).doPrintf(0xc0077b4340, 0x135e346, 0x53, 0xc029705890, 0x1, 0x1)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:1030 +0x15a fp=0xc029705800 sp=0xc029705718 pc=0x10da61a
fmt.Sprintf(0x135e346, 0x53, 0xc029705890, 0x1, 0x1, 0x12eec60, 0x1030046)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:219 +0x66 fp=0xc029705858 sp=0xc029705800 pc=0x10d35d6
github.com/grafana/tanka/pkg/kubernetes/client.ErrorNoCluster.Error(...)
        /Users/nbari/projects/go/src/github.com/grafana/tanka/pkg/kubernetes/client/errors.go:35
github.com/grafana/tanka/pkg/kubernetes/client.(*ErrorNoCluster).Error(0xc0077b2180, 0x12e0da0, 0xc0077b2180)
        <autogenerated>:1 +0xa4 fp=0xc0297058b0 sp=0xc029705858 pc=0x11ef174
fmt.(*pp).handleMethods(0xc0077b4270, 0x73, 0xc029705b01)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:624 +0x1db fp=0xc029705b20 sp=0xc0297058b0 pc=0x10d66ab
fmt.(*pp).printArg(0xc0077b4270, 0x12e0da0, 0xc0077b2180, 0x73)
        /usr/local/Cellar/go/1.14/libexec/src/fmt/print.go:713 +0x1e4 fp=0xc029705bb8 sp=0xc029705b20 pc=0x10d6d74
```

**Comment by sh0rez** [comment]
Hi, you experienced an issue with printing an actual error – This will be fixed in #239.

It should have said:

```
connecting to Kubernetes: finding usable context: no cluster that matches the apiServer `` was found. Please check your $KUBECONFIG
```

This means you did not add the address of your cluster to `environments/default/spec.json`.
You can do so by:

```bash
# use what kubectl currently uses:
$ tk env set environments/default --server-from-context=$(kubectl config current-context)

# another server:
tk env set environments/default --server=https://foobar.cluster:6443
```

**Comment by nbari** [comment]
Hi @sh0rez , thanks, I only added the IP address no the protocol`https`, probably a condition could be added that if no protocol is defined defaults to `https`
